In [1]:
%%time
from moon.linear_model import MultipleLinearRegression
import numpy as np

np.random.seed(666)
x = 2 * np.random.random(size=100)
y = x * 6. + 9. + np.random.normal(size=100)
X = x.reshape(-1, 1)

regression = MultipleLinearRegression()
regression.fit_gradient_descent(X,y)

Wall time: 170 ms


In [2]:
print(regression.coef_)

[6.0070637]


In [3]:
print(regression.intercept_)

9.02145676030123


使用代数的方法求解后，运行时间为：Wall time: 153 ms
使用向量后，运行时间为：Wall time: 325 ms   [6.0070637]
9.02145676030123